src/replication/ng/execution/execution.go

```diff
@@ -54,3 +57,130 @@ type Manager interface {
 	// Get the log of one specific task
 	GetTaskLog(int64) ([]byte, error)
 }
+
+// DefaultManager ..
+type DefaultManager struct {
+
+}
+
+// NewDefaultManager ...
+func NewDefaultManager() (Manager, error)  {
+	return &DefaultManager{}, nil
+}
+
+// Create a new execution
+func (dm *DefaultManager) Create(execution *models.Execution) (int64, error) {
+	return dao.AddExecution(execution)
+}
+
+// List the summaries of executions
+func (dm *DefaultManager) List(queries ...*models.ExecutionQuery) (int64, []*models.Execution, error) {
+	total, err := dao.GetTotalOfExecutions(queries...)
+	if err != nil {
+		return 0, nil, err
+	}
+
+	executions, err := dao.GetExecutions(queries...)
+	if err != nil {
+		return 0, nil, err
+	}
+	return total, executions, nil
+}
+
+// Get the specified execution
+func (dm *DefaultManager) Get(id int64) (*models.Execution, error) {
+	return dao.GetExecution(id)
+}
+
+// Update ...
+func (dm *DefaultManager) Update(execution *models.Execution, props ...string) error {
+	n, err := dao.UpdateExecution(execution, props...)
+	if err != nil {
+		return err
+	}
+	if n == 0 {
+		return fmt.Errorf("Execution not found error: %d ", execution.ID)
+	}
+	return nil
+}
+
+// Remove the execution specified by the ID
+func (dm *DefaultManager) Remove(id int64) error {
+	return dao.DeleteExecution(id)
+}
+
+// RemoveAll executions of one policy specified by the policy ID
+func (dm *DefaultManager) RemoveAll(policyID int64) error {
+	return dao.DeleteAllExecutions(policyID)
+}
+
+// CreateTask used to create a task
+func (dm *DefaultManager) CreateTask(task *models.Task) (int64, error) {
+	return dao.AddTask(task)
+}
+
+// ListTasks list the tasks according to the query
+func (dm *DefaultManager) ListTasks(queries ...*models.TaskQuery) (int64, []*models.Task, error) {
+	total, err := dao.GetTotalOfTasks(queries...)
+	if err != nil {
+		return 0, nil, err
+	}
+
+	tasks, err := dao.GetTasks(queries...)
+	if err != nil {
+		return 0, nil, err
+	}
+	return total, tasks, nil
+}
+
+// GetTask get one specified task
+func (dm *DefaultManager) GetTask(id int64) (*models.Task, error) {
+	return dao.GetTask(id)
+}
+
+// UpdateTask ...
+func (dm *DefaultManager) UpdateTask(task *models.Task, props ...string) error {
+	n, err := dao.UpdateTask(task, props ...)
+	if err != nil {
+		return err
+	}
+	if n == 0 {
+		return fmt.Errorf("Task not found error: %d ", task.ID)
+	}
+	return nil
+}
+
+// UpdateTaskStatus ...
+func (dm *DefaultManager) UpdateTaskStatus(taskID int64, status string, statusCondition ...string) error {
+	n, err := dao.UpdateTaskStatus(taskID, status, statusCondition ...)
+	if err != nil {
+		return err
+	}
+	if n == 0 {
+		return fmt.Errorf("Update task status failed %d: -> %s ", taskID, status)
```

Comment:
The error message should be same with others: `fmt.Errorf("Task not found error: %d ", task.ID)`?

Reply:
condition could be triggered not only by task not found.  actually error is always not null when n equals 0, so this condition will never reached.